R/xsVisualise.R

```diff
@@ -10,21 +10,24 @@ setMethod("plotEICs", "xsAnnotate", function(object,
   smpls <- unique(object@psSamples[pspec])
            
   xeic  <- new("xcmsEIC");
+  xeic@eic <- vector("list", length(pspec))
   xeic@rtrange <- matrix(nrow=length(pspec), ncol=2)
   xeic@mzrange <- matrix(nrow=length(pspec), ncol=2)
   #iterator for ps-grp
-  pcpos <- 1;
+  cnt <- 0;
   #one second overlap
   rtmargin <- 1; 
   
   for (a in seq(along=smpls)) { ## sample-wise EIC collection
     #read rawData into one xcmsRaw
     xraw <- xcmsRaw(object@xcmsSet@filepaths[smpls[a]], profmethod=method)
+    xraw@scantime <- object@xcmsSet@rt$corrected[[smpls[a]]]
 
     pspecS <- pspec[which(object@psSamples[pspec] == smpls[a])]
     ## getting ALL peaks from the current sample (not that bad)
     peaks <- CAMERA:::getPeaks(object@xcmsSet, smpls[a])  
-    eic   <- lapply (pspecS, function(pc) {
+    invisible(lapply (pspecS, function(pc) {
+      cnt <<- cnt + 1
```

Comment:
`<<-` immediately jumps at me. Any chance to avoid that ?

Reply:
Not sure why this would be an issue, becuase as I understand it <<- applies to the parent environment, which in this case is PlotEICs() where cnt is assigned at line 17 (and so not to the the global environment)?